core/src/main/java/org/mule/runtime/core/internal/message/DefaultMessageBuilder.java

```diff
@@ -122,34 +125,34 @@ public DefaultMessageBuilder(org.mule.runtime.api.message.Message message) {
 
   @Override
   public InternalMessage.Builder nullValue() {
-    this.payload = new TypedValue(null, resolveDataType(null));
+    this.payload = new TypedValue<>(null, resolveDataType(null));
     return this;
   }
 
   @Override
   public InternalMessage.Builder value(Object value) {
-    this.payload = new TypedValue(value, resolveDataType(value));
+    this.payload = new TypedValue<>(value, resolveDataType(value));
     return this;
   }
 
   @Override
-  public InternalMessage.CollectionBuilder mediaType(MediaType mediaType) {
+  public InternalMessage.MapBuilder mediaType(MediaType mediaType) {
```

Comment:
why remove it from the collection but add it to the map?

Reply:
`badmine`